### PR TITLE
ARROW-6984: [C++] Update LZ4 to 1.9.2 for CVE-2019-17543

### PR DESCRIPTION
**Pull request by kszucs**

### CODE REVIEW

**Comment by kszucs** [comment]
Linking [fails](https://travis-ci.org/kszucs/arrow/jobs/602437500#L1723) despite that `-fPIC` is [passed](https://github.com/apache/arrow/blob/master/cpp/build-support/build-lz4-lib.sh#L20) to the build script.

File: cpp/thirdparty/versions.txt [hunk 1]
```diff
@@ -35,7 +35,8 @@ GLOG_VERSION=v0.3.5
 GRPC_VERSION=v1.20.0
 GTEST_VERSION=1.8.1
 JEMALLOC_VERSION=5.2.1
-LZ4_VERSION=v1.8.3
+# Need post v1.9.2 version because of https://nvd.nist.gov/vuln/detail/CVE-2019-17543
```
Comment (pitrou) [inline cpp/thirdparty/versions.txt]: No need to add this comment IMHO.

**Review by pitrou — APPROVED** [review]
LGTM on the principle.

**Comment by pitrou** [comment]
@ursabot crossbow build

**Comment by pitrou** [comment]
@ursabot crossbow

**Comment by pitrou** [comment]
@ursabot crossbow submit

**Comment by pitrou** [comment]
@ursabot crossbow submit -g wheel linux

**Comment by pitrou** [comment]
@ursabot crossbow submit -g wheel -g linux

**Pull request closed without merging**